tests/rptest/tests/datalake/datalake_services.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from typing import Optional
+from rptest.clients.rpk import RpkTool, TopicSpec
+from rptest.services.spark_service import SparkService
+from rptest.services.trino_service import TrinoService
+from rptest.tests.datalake.iceberg_rest_catalog import IcebergRESTCatalogTest
+from rptest.services.kgo_verifier_services import KgoVerifierProducer
+from ducktape.utils.util import wait_until
+
+
+class DatalakeServicesBase(IcebergRESTCatalogTest):
+    """All inclusive base class for implementing datalake tests. Includes the
+    boiler plate to enable datalake and bring up related services.
```

Comment:
Just curious, do you actually expect tests to use both Spark and Trino? I would've thought we'd parameterize tests by query engine after encapsulating a set of well-defined operations per service. Are there things that Spark gives you as a test author that Trino doesn't have and vice versa?

Reply:
> Are there things that Spark gives you as a test author that Trino doesn't have and vice versa?

I also think parameterization is ideal (and the end goal).. but there are SQL grammar differences between Trino/Spark that are making things difficult (eg: create table in spark requires "USING ICEBERG" whereas Trino takes it from the connector config by default.. timestamps are differently named in both engines), so there has to be a shim layer that can normalize things (probably extract the most commonly used test patterns and translate to engine specific outputs), hoping it could be a follow up once we establish the e2e coverage.

Reply:
Yea my thoughts exactly